modules/core/src/utils/metrics-helper.js

```diff
@@ -48,17 +48,17 @@ function getTimeSeriesForStream(streamName, metadata, stream, target) {
  * @param streams array of streams data
  * @returns {Array} array of time series data
  */
-export function getTimeSeries({metadata = {}, streamNames, streams}) {
+export function getTimeSeries({streamMetadata = {}, streamNames, streams}) {
```

Comment:
Another situation where using `streamsMetadata` to refer to the map of all streams' metadata would disambiguate from the previously named `streamMetadata ` (i.e. the metadata of only one stream)

Throughout the PR, I see `streamMetadata` either refer to a single stream's metadata or to all streams' metadata. Renaming might help to disambiguate.

Also I would update the docstring from `logMetadata ` to `streamsMetadata`